CHANGELOG.md

```diff
@@ -66,6 +66,7 @@
 - Move `syntax_tests` into the `tests` folder. https://github.com/rescript-lang/rescript-compiler/pull/7090 https://github.com/rescript-lang/rescript-compiler/pull/7097
 - Capitalize runtime filenames. https://github.com/rescript-lang/rescript-compiler/pull/7110
 - Build mocha tests as esmodule / .mjs. https://github.com/rescript-lang/rescript-compiler/pull/7115
+- Remove Dict.t and use dict everywhere. https://github.com/rescript-lang/rescript-compiler/pull/7136
```

Comment:
@zth The question is if this should be classified as breaking or not.

It is potentially breaking for people who were using @rescript/core previously. But not for people who were using ReScript 11 without Core.

Reply:
Right. So one alternative is to let the alias remain but deprecate it and not reference it anywhere, not even in the same module. Might be the best way forward?

Reply:
Another way to think about it:

Such a type alias is only problematic if it is used in the function signatures, because then it seems to creep into hover help, .resi files etc.
Just having the type t to indicate what it is that the module's functions act on (but directly using the primitive type in the function signatures) probably doesn't hurt? So another approach would be to *add* it to modules where it is missing.
An advantage of that would be usage in functors, e.g., if a functor wants a module with `type t` and function `map` you could pass both `List` and `Array` then.

We have more such inconsistencies currently, e.g., there is `type t = char` in the `Char` module, but no `type t = bigint` in the `BigInt` module.

Reply:
@zth Let's do the `Dict.t`, `List.t`, etc. cleanup separately once we have decided how to proceed with that.

I have updated the PR once more, please re-check/approve.